components/AuthView/AuthView.js

```diff
@@ -0,0 +1,373 @@
+      this.state.email.length > 0
+    ) {
+      try {
+        const registeredUser = await fetch(`${BACKEND_URL}/auth/register`, {
+          method: 'post',
+          body: JSON.stringify({
+            username: this.state.username.toLowerCase(),
+            password: this.state.password,
+            email: this.state.email,
+            id: '',
+          }),
+          credentials: 'include',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+        if (registeredUser) {
+          const registeredUserJSON = await registeredUser.json();
+          if (registeredUserJSON.status === '500' || !registeredUserJSON.data) {
+            this.setState({regError: 'User Already Exists'});
+          } else {
+            console.log(registeredUserJSON);
+            await AsyncStorage.setItem(
+              'username',
+              registeredUserJSON.data.username,
+            );
+            await AsyncStorage.setItem(
+              'userId',
+              registeredUserJSON.data.id.toString(),
+            );
+            const storedUsername = await AsyncStorage.getItem('username');
+            const storedId = await AsyncStorage.getItem('userId');
+            const storedData = {
+              username: storedUsername,
+              id: storedId,
+            };
+            this.props.navigation.navigate('Home', {userData: storedData});
+          }
+        } else {
+          this.setState({regError: 'User Already Exists'});
+        }
+      } catch (err) {
+        console.error(`Register Failed!!! ${err}`);
+      }
+    } else {
+      this.setState({regError: 'Please Correct Errors'});
+    }
+  };
+
+  checkForUser = async () => {
+    try {
+      const storedUsername = await AsyncStorage.getItem('username');
+      const storedId = await AsyncStorage.getItem('userId');
+      const storedData = {
+        username: storedUsername,
+        id: storedId,
+      };
+      if (storedData.username && storedData.id) {
+        const user = await fetch(`${BACKEND_URL}/auth/session`, {
+          method: 'post',
+          body: JSON.stringify({
+            username: storedData.username.toLowerCase(),
+            password: '',
+            id: storedData.id,
+          }),
+          credentials: 'include',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+        if (user) {
+          const userJSON = await user.json();
+          if (userJSON.data) {
+            this.props.navigation.navigate('Home', {
+              userData: storedData,
+            });
+          }
+        }
+      }
+    } catch (err) {
+      console.error(`Checking For User Failed!!! ${err}`);
+    }
+  };
+
+  render() {
+    return (
+      <TouchableWithoutFeedback
+        onPress={() => {
+          dismissKeyboard();
+        }}>
+        <View style={styles.container}>
+          {this.state.register ? null : (
```

Comment:
Since these two blocks display entirely different content, I'd suggest either breaking them into a `renderAuthedState()/renderUnauthedState()` method just to fully separate them (they could also be separate components, which would require passing some state/callbacks in. My first step is usually to start with render methods)